ooni/templates/httpt.py

```diff
@@ -122,11 +122,28 @@ def addToReport(self, request, response=None, response_body=None, failure_string
 
             failure (instance): An instance of :class:twisted.internet.failure.Failure
         """
+        def _representHeaders(headers):
+            represented_headers = {}
+            for name, value in headers.getAllRawHeaders():
+                represented_headers[name] = value[0]
+            return represented_headers
+
+        def _representBody(body):
+            # XXX perhaps add support for decoding gzip in the future.
+            try:
+                body.replace('\0', '')
```

Comment:
Shouldn't the parser encoding the original json encode the null byte as `\\u0000`?

To test this hypothesis, here's what I tried:

```
$ python
Python 2.7.10 (default, Oct 14 2015, 16:09:02) 
[GCC 5.2.1 20151010] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> s = "helo\0admin"
>>> s
'helo\x00admin'
>>> json.dumps(s)
'"helo\\u0000admin"'
>>>

$ python3
Python 3.4.3+ (default, Oct 14 2015, 16:03:50) 
[GCC 5.2.1 20151010] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> s = "helo\0admin"
>>> s
'helo\x00admin'
>>> json.dumps(s)
'"helo\\u0000admin"'
>>>

$ nodejs
> s = "helo\0admin"
'helo\u0000admin'
> JSON.stringify(s)
'"helo\\u0000admin"'
>
```